README with enhanced install instructions for protobuf (#715)
OpenSimulationInterface/open-simulation-interface: doc/setup/installing_prerequisites.adoc

ifndef::include-only-once[]
:root-path: ../
include::{root-path}_config.adoc[]
endif::[]
[#top-installing-osi-prerequisits]
= Installing protobuf for static / dynamic linking

== Prerequisites

* You have installed _cmake_ (version 3.10.2 or higher).
* You must have super user privileges.

Static linking is the recommended way to package OSI FMUs for sharing with others to be integrated in a co-simulation: They should have minimal external dependencies, in order to ensure maximum portability.
Fo other use cases like first trials to get started with OSI, dynamic linking could be an option, as well.

== Windows

> **Dynamic Linking (NOT RECOMMENDED)**

Reviewer: I would recommend using description lists instead of manually formatting as bold:

```suggestion
Dynamic Linking (NOT RECOMMENDED)::
```

Reviewer: This goes for all the following examples as well.
However, the empty line between this and its description should also be removed then.

Reviewer: Ideally: yes. However, this could be done in a separate cleanup issue then.

It is not a required step ofc.  It's just that I personally feel that using the included functionality of sections or description lists for this type of section declarations is better.

However, if you would like to stick to the current styling (i.e. not numbered, same font size, and no tabulation), it might even be the best approach.

Author: Ok, so let's put it on the agenda for the next CCB and discuss if this should be opened as a separate issue there.

Author: As decided in the CCB, I have changed the PR now, using description lists instead of manually formatting as bold in doc/setup.

Since on windows symbols are not exported per default it can be kind of annoying to deal with this during _protobuf_ header generation (see for example  https://groups.google.com/g/protobuf/c/PDR1bqRazts).

> So basically thats one more reason to use static linking.

Reviewer: Would rephrase as follows:

```suggestion
That is an important reason to use static linking.
```
